data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/PublishedNewsDisplayedPropUpgrade.java

```diff
@@ -67,34 +72,47 @@ public PublishedNewsDisplayedPropUpgrade(InitParams initParams,
     this.container = container;
   }
 
-  public int getPublishedNewsDisplayedPropCount() {
-    return publishedNewsDisplayedPropCount;
+  public int getMigratedPublishedNewsCount() {
+    return migratedPublishedNewsCount;
   }
 
   @Override
   public void processUpgrade(String oldVersion, String newVersion) {
     long startupTime = System.currentTimeMillis();
     LOG.info("Start upgrade of published news displayed property");
-
-    boolean transactionStarted = false;
-    RequestLifeCycle.begin(container);
+    ExoContainerContext.setCurrentContainer(container);
+    List<String> newsTargetMetadatas = metadataService.getMetadatas(NewsTargetingService.METADATA_TYPE.getName(), 0).stream()
+        .map(newsTargetMetadata -> String.valueOf(newsTargetMetadata.getId()))
+        .collect(Collectors.toList());
+    List<MetadataItemEntity> newsTargetsMetadataItems = new ArrayList<>();
+    RequestLifeCycle.begin(entityManagerService);
     EntityManager entityManager = entityManagerService.getEntityManager();
     try {
-      if (!entityManager.getTransaction().isActive()) {
-        entityManager.getTransaction().begin();
-        transactionStarted = true;
-      }
-      List<Metadata> newsTargetMetadatas = metadataService.getMetadatas(NewsTargetingService.METADATA_TYPE.getName(), 0);
-
-      List<MetadataItemEntity> newsTargetsMetadataItems = new ArrayList<>();
-      for (Metadata newsTargetMetadata : newsTargetMetadatas) {
-        StringBuilder newsTargetsMetadataItemsQueryStringBuilder = new StringBuilder("SELECT * FROM SOC_METADATA_ITEMS WHERE METADATA_ID = '");
-        newsTargetsMetadataItemsQueryStringBuilder.append(newsTargetMetadata.getId() + "'");
-        Query newsTargetsMetadataItemsQuery = entityManager.createNativeQuery(newsTargetsMetadataItemsQueryStringBuilder.toString(), MetadataItemEntity.class);
-        newsTargetsMetadataItems.addAll(newsTargetsMetadataItemsQuery.getResultList());
+      StringBuilder newsTargetsMetadataItemsQueryStringBuilder = new StringBuilder("SELECT * FROM SOC_METADATA_ITEMS WHERE METADATA_ID IN ");
+      newsTargetsMetadataItemsQueryStringBuilder.append("(" + String.join(",", newsTargetMetadatas) + ")");
+      Query newsTargetsMetadataItemsQuery = entityManager.createNativeQuery(newsTargetsMetadataItemsQueryStringBuilder.toString(), MetadataItemEntity.class);
+      newsTargetsMetadataItems = newsTargetsMetadataItemsQuery.getResultList();
+    } catch (Exception e) {
+      if (entityManager.getTransaction().isActive() && entityManager.getTransaction().getRollbackOnly()) {
+        entityManager.getTransaction().rollback();
```

Comment:
No need to rollback neither manage the transaction because you didn't made a change on database.
Whereas, when an exception occurs here, you will need to throw an `Exception` and terminate the upgrade process, else the Upgrade Plugin will not run again and will be set as processed.